engines/tinsel/dialogs.cpp

```diff
@@ -4994,45 +5224,48 @@ void Dialogs::syncInvInfo(Common::Serializer &s) {
  */
 // Note: the SCHANDLE type here has been changed to a void*
 void Dialogs::RegisterIcons(void *cptr, int num) {
-	_numObjects = num;
-	_invObjects = (INV_OBJECT *)cptr;
+	int numObjects = num;
+	INV_OBJECT *invObjects = (INV_OBJECT *)cptr;
 
 	if (TinselVersion == 0) {
```

Comment:
We can get rid of the whole block here if we have a version of `InstantiateInventoryObjects()` for V0. Check the suggestion inside that function